tools/purge/conffile.cc

```diff
@@ -47,9 +47,14 @@
 #include <cstdlib>
 #include <cstring>
 #include <fstream>
-#include <sys/types.h>
 #include <memory.h>
 
+#if HAVE_REGEX_H
+#include <regex.h>
+#endif
+
+#include <sys/types.h>
```

Comment:
Note the detail from now removed compat/GnuRegex.h :
```
/* POSIX says that <sys/types.h> must be included (by the caller) before
 * <regex.h>.  */
```

Reply:
Thanks for catching. According to https://pubs.opengroup.org/onlinepubs/009695399/basedefs/regex.h.html , that' in order to get size_t.

I'll add sys/types.h before regex.h whenever included.

Reply:
It's probably not worth keeping these sys/types.h includes at all. squid.h should be pulling it in early already.
With the switch to std::regex coming shortly I don't think it matters if done the way you have applied already, so not complaining about that.

Reply:
> ```C++
> /* POSIX says that <sys/types.h> must be included (by the caller) before
>  * <regex.h>.  */
> ```

That assertion used to be correct, but it became false since POSIX 1003.1-2001. See https://sourceware.org/bugzilla/show_bug.cgi?id=2534

Modern regex.h includes the necessary dependencies itself as needed, of course. For example: https://codebrowser.dev/glibc/glibc/posix/regex.h.html#23